app/js/authentication_controller.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+/* global SumoDB, User, Utils */
+
+(function(exports) {
+  var USER_CREDENTIALS_KEY = 'user_credentials';
+
+  function authenticate(evt) {
+    evt.preventDefault();
+
+    var form = document.getElementById('login');
+    var username = form.elements['username'].value;
+    var password = form.elements['password'].value;
+
+    var promise = SumoDB.authenticate_user(username, password);
+
+    Utils.toggle_spinner();
+
+    promise.then(function(response) {
```

Comment:
Given that it's an event handler, there is no real point returning this. But in general, yes, we want to return promises.